backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -190,10 +190,11 @@ private void checkIfEntityAlreadyExists(Session session, Project project) {
   @Override
   public Project insertProject(Project project, UserInfo userInfo)
       throws InvalidProtocolBufferException {
-    createRoleBindingsForProject(project, userInfo);
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      Transaction transaction = session.beginTransaction();
       checkIfEntityAlreadyExists(session, project);
+      createRoleBindingsForProject(project, userInfo);
+
+      Transaction transaction = session.beginTransaction();
```

Comment:
due to this. MDB creates the RoleBinding again when the project already exists with the same name because already exists condition is below the `createRoleBindingsForProject`.